runtime/autoload/provider/clipboard.vim

```diff
@@ -8,39 +8,121 @@ function! s:try_cmd(cmd, ...)
   let out = a:0 ? systemlist(a:cmd, a:1, 1) : systemlist(a:cmd, [''], 1)
   if v:shell_error
     echo "clipboard: error: ".(len(out) ? out[0] : '')
-    return ''
+    return ['']
   endif
   return out
 endfunction
 
-if executable('pbcopy')
+let s:use_vimenc = 0
+
+if executable('xsel-vim')
+  let s:use_vimenc = 1
+  let s:copy['+'] = 'xsel-vim --vim --nodetach -i -b'
+  let s:paste['+'] = 'xsel-vim --vim -o -b'
+  let s:copy['*'] = 'xsel-vim --vim --nodetach -i -p'
+  let s:paste['*'] = 'xsel-vim --vim -o -p'
+elseif executable('pbcopy')
   let s:copy['+'] = 'pbcopy'
   let s:paste['+'] = 'pbpaste'
   let s:copy['*'] = s:copy['+']
   let s:paste['*'] = s:paste['+']
 elseif executable('xclip')
-  let s:copy['+'] = 'xclip -i -selection clipboard'
+  let s:copy['+'] = 'xclip -quiet -i -selection clipboard'
   let s:paste['+'] = 'xclip -o -selection clipboard'
-  let s:copy['*'] = 'xclip -i -selection primary'
+  let s:copy['*'] = 'xclip -quiet -i -selection primary'
   let s:paste['*'] = 'xclip -o -selection primary'
 elseif executable('xsel')
-  let s:copy['+'] = 'xsel -i -b'
+  let s:copy['+'] = 'xsel --nodetach -i -b'
   let s:paste['+'] = 'xsel -o -b'
-  let s:copy['*'] = 'xsel -i -p'
+  let s:copy['*'] = 'xsel --nodetach -i -p'
   let s:paste['*'] = 'xsel -o -p'
 else
   echom 'clipboard: No shell command for communicating with the clipboard found.'
   finish
 endif
 
+let s:owning = {'*': -1, '+': -1}
+let s:data = {'*': [], '+': []}
+
+let g:debug_owning = s:owning
+
+au JobActivity clipprovider call <SID>on_activity()
+
+function! s:on_activity()
+    if v:job_data[1] == "exit"
+        for reg in ['*', '+']
+            if s:owning[reg] == v:job_data[0]
```

Comment:
@justimk don't worry about me, I've been having my hands full with my PhD
application. ;) Im glad to see this getting resolved